doc/source/whatsnew/v1.3.0.rst

```diff
@@ -647,6 +647,7 @@ Deprecations
 - Deprecated setting :attr:`Categorical._codes`, create a new :class:`Categorical` with the desired codes instead (:issue:`40606`)
 - Deprecated behavior of :meth:`DatetimeIndex.union` with mixed timezones; in a future version both will be cast to UTC instead of object dtype (:issue:`39328`)
 - Deprecated using ``usecols`` with out of bounds indices for ``read_csv`` with ``engine="c"`` (:issue:`25623`)
+- Deprecated passing arguments as positional in :meth:`DataFrame.drop_duplicates` (except for ``subset``) and :meth:`Series.drop_duplicates` (:issue:`41485`)
```

Comment:
maybe we should do the same for Index at the same time for consistency.

Reply:
we will get `error: Signature of "drop_duplicates" incompatible with supertype "IndexOpsMixin"  [override]` otherwise

Reply:
Should I add this warning decorator to Index.drop_duplicates + test in a commit on this branch / pull request?

Reply:
Yes, that would be great, thanks @jmholzer !

Reply:
I made these changes in the commit 064268f.

However, it looks like a lot of commits that I pulled from the master branch ended up in this pull request also, I don't have enough knowledge of git to understand why this happened. Is this a problem? If so, can I fix it?

Reply:
Yeah you'll need to rebase (see https://youtu.be/hv8dhOEzQcM), currently this is showing lots of unrelated changes. Something like
```
git fetch upstream
git rebase -i upstream/master
```
and then in the interactive window choose which commits to keep/drop/fixup/edit

Reply:
Thanks for the guidance, I appreciate it.

I did the following to fix things:
1. Reverted the changes on my local branch to before the rebase.
2. Cherry-picked my commit with the changes for Index from the remote.
3. Pulled remote master changes to my local master branch.
4. Merged my local feature branch with my local master branch.
5. Force-pushed my local feature branch to the remote.

That has gotten rid of the spurious commits. I tried using a rebase instead of a merge in step 3, as you suggested, but I couldn't do it cleanly.

Are steps 3-4 acceptable for working on pandas, or is a rebase generally preferred to a merge?

Reply:
yeah that's fine, commits will get squashed anyway